primitives/io/src/lib.rs

```diff
@@ -566,7 +566,23 @@ pub trait Hashing {
 	}
 }
 
+/// Interface that provides functions to access the offchain database access.
+#[runtime_interface]
+pub trait OffchainIndex {
+	/// Write a key value pair to the offchain worker database in a buffered fashion.
+	fn set_offchain_storage(&mut self, key: &[u8], value: &[u8]) {
```

Comment:
Are the generated APIs accessible via module, like so:
```
offchain_index::set_offchain_storage
```
?

If yes, I think we could reduce duplication and just do:
```
offchain_index::set(...)
offchain_index::clear(...)
```